typescript/cli/src/deploy/core.ts

```diff
@@ -0,0 +1,434 @@
+    mergedContractAddrs,
+    multiProvider,
+  );
+
+  // 3. Deploy core contracts to local chain
+  log(`Deploying core contracts to ${local}`);
+  const coreDeployer = new HyperlaneCoreDeployer(multiProvider, ismFactory);
+  coreDeployer.cacheAddressesMap(artifacts);
+  const coreConfig = buildCoreConfigMap(owner, local, remotes, multiSigConfig);
+  const coreContracts = await coreDeployer.deploy(coreConfig);
+  artifacts = writeMergedAddresses(contractsFilePath, artifacts, coreContracts);
+  logGreen(`Core contracts deployed`);
+
+  // 4. Deploy ISM contracts to remote deployable chains
+  log(`Deploying ISMs`);
+  const ismConfigs = buildIsmConfigMap(
+    owner,
+    remotes,
+    selectedChains,
+    multiSigConfig,
+  );
+  const ismContracts: ChainMap<{ multisigIsm: DeployedIsm }> = {};
+  for (const [ismChain, ismConfig] of Object.entries(ismConfigs)) {
+    if (artifacts[ismChain].multisigIsm) {
+      log(`ISM contract recovered, skipping ISM deployment to ${ismChain}`);
+      continue;
+    }
+    log(`Deploying ISM to ${ismChain}`);
+    ismContracts[ismChain] = {
+      multisigIsm: await ismFactory.deploy(ismChain, ismConfig),
+    };
+  }
+  artifacts = writeMergedAddresses(contractsFilePath, artifacts, ismContracts);
+  logGreen(`ISM contracts deployed `);
+
+  // 5. Deploy TestRecipients to all deployable chains
+  log(`Deploying test recipient contracts`);
+  const testRecipientConfig = buildTestRecipientConfigMap(
+    selectedChains,
+    artifacts,
+  );
+  const testRecipientDeployer = new TestRecipientDeployer(multiProvider);
+  testRecipientDeployer.cacheAddressesMap(artifacts);
+  const testRecipients = await testRecipientDeployer.deploy(
+    testRecipientConfig,
+  );
+  artifacts = writeMergedAddresses(
+    contractsFilePath,
+    artifacts,
+    testRecipients,
+  );
+  logGreen(`Test recipient contracts deployed`);
+
+  log('Writing agent configs');
+  await writeAgentConfig(
+    agentFilePath,
+    artifacts,
+    local,
+    remotes,
+    multiProvider,
+  );
+  logGreen('Agent configs written');
+
+  logBlue('Deployment is complete!');
+  logBlue(`Contract address artifacts are in ${contractsFilePath}`);
+  logBlue(`Agent configs are in ${agentFilePath}`);
+}
+
+function buildIsmConfig(
+  owner: Address,
+  remotes: ChainName[],
+  multisigIsmConfigs: ChainMap<MultisigIsmConfig>,
+): RoutingIsmConfig {
+  const mergedMultisigIsmConfig: ChainMap<MultisigIsmConfig> = objMerge(
+    defaultMultisigIsmConfigs,
+    multisigIsmConfigs,
+  );
+  return {
+    owner,
+    type: ModuleType.ROUTING,
+    domains: Object.fromEntries(
+      remotes.map((remote) => [remote, mergedMultisigIsmConfig[remote]]),
+    ),
+  };
+}
+
+function buildIsmConfigMap(
+  owner: Address,
+  chains: ChainName[],
+  remotes: ChainName[],
+  multisigIsmConfigs: ChainMap<MultisigIsmConfig>,
+): ChainMap<RoutingIsmConfig> {
+  return Object.fromEntries(
+    chains.map((chain) => {
```

Comment:
Is this not just objMap?

Reply:
Not quite b.c. we're looping off of `chains`, which is an array, not `multisigIsmConfigs`, which may not contain all chains.